tooling/debugger/src/context.rs

```diff
@@ -276,12 +314,24 @@ impl<'a, B: BlackBoxFunctionSolver<FieldElement>> DebugContext<'a, B> {
         if ip >= self.get_opcodes().len() {
             None
         } else {
-            let opcode_location = if let Some(ref solver) = self.brillig_solver {
-                OpcodeLocation::Brillig { acir_index: ip, brillig_index: solver.program_counter() }
-            } else {
-                OpcodeLocation::Acir(ip)
-            };
-            Some(DebugLocation { circuit_id: self.current_circuit_id, opcode_location })
+            let (opcode_location, brillig_function_id) =
+                if let Some(ref solver) = self.brillig_solver {
+                    let function_id = solver.function_id;
+                    (
+                        OpcodeLocation::Brillig {
+                            acir_index: ip,
+                            brillig_index: solver.program_counter(),
+                        },
+                        Some(function_id),
+                    )
+                } else {
+                    (OpcodeLocation::Acir(ip), None)
+                };
+            Some(DebugLocation {
+                circuit_id: self.current_circuit_id,
+                brillig_function_id,
+                opcode_location,
+            })
```

Comment:
Now that we have another field in `Debuglocation` I don't think it pays off to avoid the duplication of its creation. The only thing that's the same both cases is the `circuit_id`

IMHO this piece of code is simpler and _I find it_ more easy to reason about

```rust
    pub(super) fn get_current_debug_location(&self) -> Option<DebugLocation> {
        let ip = self.acvm.instruction_pointer();
        if ip >= self.get_opcodes().len() {
            None
        } else {
            let debug_location = match self.brillig_solver {
                Some(ref solver) => DebugLocation {
                    circuit_id: self.current_circuit_id,
                    brillig_function_id: Some(solver.function_id),
                    opcode_location: OpcodeLocation::Brillig {
                        acir_index: ip,
                        brillig_index: solver.program_counter(),
                    },
                },
                None => DebugLocation {
                    circuit_id: self.current_circuit_id,
                    opcode_location: OpcodeLocation::Acir(ip),
                    brillig_function_id: None,
                },
            };
            Some(debug_location)
        }
    }
```